app/server/apiGatewayDiscovery.ts

```diff
@@ -213,3 +213,29 @@ export const getContactUsAPIHostAndKey = async () => {
 
   return { host: `https://${host}/${stage}/`, apiKey };
 };
+
+/**
+ * Used to fail riff raff deployment. Uses an arbitrarily selected service layer api as a check.
+ * Memoised in the sense of credentials fetching happens only once on deployment.
+ */
+export const memoisedApiGatewayAuthForHealthcheck = () => {
+  const memoisedHostKeyPair = getHostAndApiKeyForStack(
+    "membership",
+    "holiday-stop-api",
+    "PROD"
+  );
```

Comment:
one refactor that could be made (probably not for this PR) would be to refactor the `apiNames` const further up in the list to be an array of tuples/objects containing the `stackPrefix` and `apiName` (which are currently params to the `getHostAndApiKeyForStack` function, that way for extra confidence we could perform this  check for all the known APIs (rather than an arbitrary one).

Reply:
If one fails, I think it is likely all of them would fail. But having a list of all the api names might be generally useful even beyond the healthcheck. Separate PR.